### PR TITLE
[qna] regen with latest and implement design changes

### LINKED ISSUE #20806: [qna] regenerate with new swagger changes
regenerate qna again when this swagger PR gets merged https://github.com/Azure/azure-rest-api-specs/pull/16063/files

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #20806

- [x] regenerated with hash 68e7988aba481206f08019d3efb585683d5bc577
- [x] rename options models to method name + options
- [x] added kwarg `default_language` onto client
- [x] `StrictFilters` -> `QueryFilters`
- [x] `CompoundOperationKind` -> `LogicalOperationKind`
- [x] Removed all traces of string index type, and pass in the unicode value
- [x] `metadata` is now a list of tuples
- [x] Made the model positional input to the methods positional only (for all intents and purposes), and documentation shows its name as `options`

### CODE REVIEW

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/swagger/README.md [hunk 1]
```diff
@@ -58,3 +58,65 @@ directive:
...
+```
+
+### Make `MetadataFilter`'s `metadata` property a list of string
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["definitions"]
+    transform: >
+        delete $["MetadataFilter"]["properties"]["metadata"]["items"]["$ref"];
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/swagger/README.md:118]: we actually want a list of tuple of string, but autorest can't handle tuples yet, so we're going to transform it into a list of list of string for easier code changes post-generation

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/operations/_operations.py [hunk 2]
```diff
@@ -95,14 +94,12 @@ async def query_knowledge_base(
...
         :paramtype ranker_type: str or ~azure.ai.language.questionanswering.models.RankerType
-        :keyword strict_filters: Filter QnAs based on give metadata list and knowledge base source names.
-        :paramtype strict_filters: ~azure.ai.language.questionanswering.models.StrictFilters
+        :keyword filters: Filter QnAs based on give metadata list and knowledge base source names.
+        :paramtype filters: ~azure.ai.language.questionanswering.models.QueryFilters
         :keyword answer_span_request: To configure Answer span prediction feature.
         :paramtype answer_span_request: ~azure.ai.language.questionanswering.models.AnswerSpanRequest
         :keyword include_unstructured_sources: (Optional) Flag to enable Query over Unstructured Sources.
         :paramtype include_unstructured_sources: bool
-        :keyword callable cls: A custom type or function that will be passed the direct response
```
Comment (iscai-msft) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/operations/_operations.py:104]: purposely removed `cls` documentation

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py [hunk 3]
```diff
@@ -36,3 +36,26 @@ def _validate_text_records(records):
...
             request_batch.append(doc)
     return request_batch
+
+def _get_positional_body(*args, **kwargs):
+    """Verify args and kwargs are valid, and then return the positional body, if users passed it in."""
+    if len(args) > 1:
+        raise TypeError("There can only be one positional argument, which is the post body of this request.")
+    if args and "options" in kwargs:
+        raise TypeError(
+            "You have already supplied the request body as a positional parameter, you can not supply it as a kwarg as well."
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py:46]: nit: would spell out kwarg as "keyword argument"

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py [hunk 4]
```diff
@@ -36,3 +36,26 @@ def _validate_text_records(records):
         else:
             request_batch.append(doc)
     return request_batch
+
+def _get_positional_body(*args, **kwargs):
+    """Verify args and kwargs are valid, and then return the positional body, if users passed it in."""
+    if len(args) > 1:
+        raise TypeError("There can only be one positional argument, which is the post body of this request.")
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py:43]: nit: should post be capitalized, POST?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/tests/test_query_knowledgebase_async.py [hunk 5]
```diff
@@ -317,9 +320,9 @@ async def test_query_knowledgebase_with_followup(self, qna_account, qna_key, qna
             )
 
             assert len(output.answers) == 2
-            confident_answers = [a for a in output.answers if a.confidence_score > 0.6]
+            confident_answers = [a for a in output.answers if a.confidence_score > 0.5]
             assert len(confident_answers) == 1
-            assert confident_answers[0].answer_span.text == "two to four hours"
+            assert confident_answers[0].answer_span.text == " two to four hours"
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/tests/test_query_knowledgebase_async.py:325]: intentional spacing?

**Review by kristapratico — APPROVED** [review]
nits, LGTM

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/tests/test_query_knowledgebase_async.py (hunk 5 above)
Comment (iscai-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/tests/test_query_knowledgebase_async.py:325]: yeah sadly the spacing the service returns is different

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py [hunk 6]
```diff
@@ -36,3 +36,27 @@ def _validate_text_records(records):
...
+
+def _verify_qna_id_and_question(query_knowledgebase_options):
+    """For query_knowledge_base we require either `question` or `qna_id`."""
+    try:
+        qna_id = query_knowledgebase_options.qna_id
+        question = query_knowledgebase_options.question
+    except AttributeError:
+        qna_id = query_knowledgebase_options.get("qna_id") or query_knowledgebase_options.get("qnaId")
+        question = query_knowledgebase_options.get("question")
+    if not (qna_id or question):
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py:59]: Should you also check that it's one or the other, not both?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_question_answering_client.py [hunk 7]
```diff
@@ -49,6 +51,7 @@ def __init__(
         self._serialize = Serializer(client_models)
         self._deserialize = Deserializer(client_models)
         self._serialize.client_side_validation = False
+        self._default_language = kwargs.pop("default_language", "en")
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_question_answering_client.py:54]: Default to empty string and let the service decide. That's what I did in .NET as well.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_question_answering_client.py [hunk 8]
```diff
@@ -27,6 +27,8 @@ class QuestionAnsweringClient(QuestionAnsweringClientOperationsMixin):
     :type endpoint: str
     :param credential: Credential needed for the client to connect to Azure.
     :type credential: ~azure.core.credentials.AzureKeyCredential
+    :keyword str default_language: Sets the default language to use for all operations.
+     Defaults to "en".
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_question_answering_client.py:31]: Should default to "" or None, whichever is appropriate here.

**Review by heaths — APPROVED** [review]

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_question_answering_client.py (hunk 8 above)
Comment (iscai-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_question_answering_client.py:31]: ah ok, so we shouldn't specify "en" as the default on our side, but rely on server default

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py (hunk 6 above)
Comment (iscai-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py:59]: ahh good point

**Pull request merged**